src/alire/alire-publish.adb

```diff
@@ -59,6 +60,40 @@ package body Alire.Publish is
    --  The following procedures share the spec, in the case in the future we
    --  need to reorder/reuse/insert them as part of a navigable workflow.
 
+   -----------------
+   -- Check_Build --
+   -----------------
+
+   procedure Check_Build (Context : in out Data) with
+     Pre => GNAT.OS_Lib.Is_Directory (Context.Tmp_Deploy_Dir.Filename);
+   --  Check that the sources we are trying to publish can be built
+
+   procedure Check_Build (Context : in out Data) is
+      --  Enter the temporary as if it were a workspace (which it has to
+      --  be, as it contains the user manifest). Auto-update should retrieve
+      --  dependencies, and since we are not repacking, there's no problem
+      --  with altering contents under alire or regenerating the lock file.
+      Guard : Directories.Guard
+        (Directories.Enter (Context.Tmp_Deploy_Dir.Filename))
+        with Unreferenced;
+   begin
+      --  We need the alire folder, which usually will not be among sources
+      if not Ada.Directories.Exists (Paths.Working_Folder_Inside_Root) then
+         Ada.Directories.Create_Directory (Paths.Working_Folder_Inside_Root);
+      end if;
+
+      --  Unfortunately, building is one of the parts that still are hard to
+      --  extricate from Alr.*, so for now the simplest solution is to spawn
+      --  a child alr.
+      OS_Lib.Subprocess.Checked_Spawn
```

Comment:
I have projects that cannot be built standalone, GESTE for example. So it should be possible to ignore this check.

Reply:
I was thinking the same if only because some builds take a long time. For now I'll add a switch to turn this entirely off. Later, I've thought that here we should run any test actions instead, defaulting to this.